pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -221,6 +242,11 @@ private BrokerResponse handleRequest(long requestId, String query,
     return brokerResponse;
   }
 
+  private boolean hasTableAccess(RequesterIdentity requesterIdentity, RelNode relRoot) {
+    Set<String> tableNames = new HashSet<>(RelOptUtil.findAllTableQualifiedNames(relRoot));
+    return _accessControlFactory.create().hasAccess(requesterIdentity, tableNames);
+  }
```

Comment:
Is looking at the `RelRoot` actually necessary to get the list of table names? Later in the code i see that the `QueryPlan` has the list of tables:

```
List<String> tableNames = queryPlan.getStageMetadataMap().get(entry.getKey()).getScannedTables();
```

Perhaps we can reuse the above rather than trying to expose the `RelRoot`?

Optionally do you think we should push down this validation to the actual `explainPlan` or `planQuery` methods themselves instead?

Reply:
> Later in the code i see that the QueryPlan has the list of tables:

I did see this way. But didn't go with this approach because it wouldn't work for explain plan queries. 


> Optionally do you think we should push down this validation to the actual explainPlan or planQuery methods themselves instead if you really need access to the RelRoot to get the table names?

Yes, this is an option (and the only one available now I guess). But the cons is that we'll have to pass a bunch of context like _accessControlFactory, _brokerMetrics, etc to QueryContext.

Reply:
Yeah i think maybe there isn't an option. Best to look into how to pass this information down (at least for `explainPlan` where we can't utilize the `QueryPlan`). One more option might be to have `explainPlan` return an object containing the `RelRoot` along with the string?

Reply:
I don't think it'll be cleaner to pass down the this information as this will affect error handling and lead to more info passed down in the future. 
I am already exploring the idea of returning RelRoot for explain plan. Will update this PR.

Reply:
I've updated the PR. Please take a look to see if this approach works. Do let me know if you can think of a better way!

Reply:
approach lgtm, thanks for addressing this!